src/core_editor/line_buffer.rs

```diff
@@ -541,26 +541,67 @@ mod test {
         assert_eq!(expected_line_buffer, line_buffer);
     }
 
-    #[test]
-    #[ignore] // Note: Not sure if this is the intended behaviour
-    fn uppercase_word_works_when_one_last_index() {
-        let mut line_buffer = buffer_with("This is a test");
-        line_buffer.uppercase_word();
+    #[rstest]
+    // FIXME: capitalize_char move insertion point to the end
+    #[case("This is a test", (0, 13), "This is a tesT", (0, 14))]
+    // FIXME: capitalize_char move insertion point to the end
+    #[case("This is a test", (0, 10), "This is a Test", (0, 14))]
+    fn capitalize_char_works(
+        #[case] input: &str,
+        #[case] in_location: (usize, usize),
+        #[case] output: &str,
+        #[case] out_location: (usize, usize),
+    ) {
+        let mut line_buffer = buffer_with(input);
+        line_buffer.set_insertion_point(in_location.0, in_location.1);
+        line_buffer.capitalize_char();
 
-        let expected_line_buffer = buffer_with("This is a TEST");
+        let mut expected = buffer_with(output);
+        expected.set_insertion_point(out_location.0, out_location.1);
 
-        assert_eq!(expected_line_buffer, line_buffer);
+        assert_eq!(expected, line_buffer);
     }
 
-    #[test]
-    fn uppercase_word_works() {
-        let mut line_buffer = buffer_with("This is a test");
-        line_buffer.move_word_left();
+    #[rstest]
+    // FIXME: uppercase_word does not work when on penultimate location
+    // #[case("This is a test", (0, 13), "This is a TEST", (0, 13))]
+    // FIXME: uppercase_word updates the insertion point location
+    #[case("This is a test", (0, 10), "This is a TEST", (0, 14))]
```

Comment:
- Should the cursor position change
- This updates the words after the point instead of the word